agent/api/ecsclient/client_test.go

```diff
@@ -250,13 +256,30 @@ func TestRegisterContainerInstance(t *testing.T) {
 		resource, ok := findResource(req.TotalResources, "PORTS_UDP")
 		assert.True(t, ok, `Could not find resource "PORTS_UDP"`)
 		assert.Equal(t, "STRINGSET", *resource.Type, `Wrong type for resource "PORTS_UDP"`)
-		assert.Equal(t, len(capabilities)+1, len(req.Attributes), "Wrong length of Attributes")
-		for i, _ := range capabilities {
-			assert.NotNil(t, req.Attributes[i].Name, "nil name for attribute")
-			assert.Equal(t, capabilities[i], *req.Attributes[i].Name)
+		assert.Equal(t, len(capabilities)+3, len(req.Attributes), "Wrong length of Attributes")
+		found_ostype, found_cap1, found_cap2, found_attr1, found_attr2, found_unknown := 0, 0, 0, 0, 0, 0
+		for i := range req.Attributes {
```

Comment:
(copying old comment because it's now collapsed by default)

I think it'd be clearer to construct the expected attributes and then do `assert.Equal`.

```go
expectedAttributes := []*ecs.Attribute{
	{Name: "ecs.os-type", Value: api.OSType},
	// etc
}
```

If you're worried about order, you can try using [`assert.Contains`](https://godoc.org/github.com/stretchr/testify/assert#Contains), something like the following (I haven't tried this yet):

```go
assert.Contains(t, req.Attributes, &ecs.Attribute{Name: "ecs.os-type", Value: api.OSType}, "Attributes should contain the OS type")
```